### PR TITLE
feat(ingest): project adapter v1 envelopes into transcript and tool extraction

### PR DESCRIPTION
**Pull request by kunickiaj**
## Description
Introduces adapter-envelope projection in ingest so v1 adapter events participate in prompt/tool extraction without breaking legacy OpenCode event handling. Also adds adapter schema/fixtures and regression coverage for adapter prompt/tool mapping edge cases.

## Type of Change

- [x] 🚀 Feature (new functionality)
- [ ] 🐛 Bug fix (fixes an issue)
- [ ] 📚 Documentation (docs-only change)
- [ ] 🔧 Maintenance (refactor, chore, CI, etc.)
- [x] 🧪 Testing (test-only changes)

## Testing

- [x] Tests pass locally (`pytest`)
- [x] Added/updated tests for changes
- [x] Manually verified changes work as expected

Commands run:
- `uv run pytest tests/test_plugin_ingest.py`
- `uv run ruff check codemem/plugin_ingest.py tests/test_plugin_ingest.py`

## Checklist

- [x] Code follows project style (`ruff check` and `ruff format` pass)
- [x] Self-review completed
- [x] Documentation updated (if needed)
- [x] No new warnings introduced

### CODE REVIEW

**Comment by kunickiaj** [comment]
* **#134** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/134?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#133** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/133?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#132** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/132?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#131** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/131?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#130** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/130?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#129** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/129?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#128** <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/128?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/kunickiaj/codemem/128?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: codemem/plugin_ingest.py [hunk 1]
```diff
@@ -262,6 +299,109 @@ def _extract_prompts(events: Iterable[dict[str, Any]]) -> list[dict[str, Any]]:
...
+    if not isinstance(tool_input, dict):
+        tool_input = {}
+    tool_error = payload.get("tool_error")
+    if tool_error is None and payload.get("status") == "error":
+        tool_error = payload.get("error")
+    return {
+        "type": "tool.execute.after",
+        "tool": payload.get("tool_name"),
+        "args": tool_input,
+        "result": payload.get("tool_output"),
```
[the remark this answers is not in the document]
Comment (kunickiaj) [reply codemem/plugin_ingest.py]: Fixed on this PR in `ac0be60`: adapter tool projection now falls back to `payload.output` when `payload.tool_output` is absent, so redacted OpenCode fixture output is preserved for observer context.

File: codemem/plugin_ingest.py [hunk 2]
```diff
@@ -247,6 +269,21 @@ def _normalize_paths(paths: list[str], repo_root: str | None) -> list[str]:
...
+        if adapter and adapter.get("event_type") == "prompt":
+            payload = adapter.get("payload")
+            if not isinstance(payload, dict):
+                continue
+            prompt_text = str(payload.get("text") or "").strip()
+            if prompt_text:
+                prompts.append(
+                    {
+                        "prompt_text": prompt_text,
+                        "prompt_number": payload.get("prompt_number"),
```
[the remark this answers is not in the document]
Comment (kunickiaj) [reply codemem/plugin_ingest.py]: Fixed on this PR in `ac0be60`: prompt extraction now uses adapter `payload.prompt_number` with fallback to legacy `event.prompt_number` when adapter omits it, preserving turn grouping traceability.

**Comment by kunickiaj** [comment]
### Merge activity

* **Mar 3, 1:13 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/kunickiaj/codemem/128).
* **Mar 3, 1:13 AM UTC**: @kunickiaj merged this pull request with [Graphite](https://app.graphite.com/github/pr/kunickiaj/codemem/128).

**Comment by kunickiaj** [comment]
AFM wrap status while stack is being merged:

- Updated bead tracking:
  - Closed: codemem-afm.1, codemem-afm.2, codemem-afm.3
  - In progress: codemem-afm, codemem-afm.4
  - Open follow-ups: codemem-afm.5 (transcript fallback), codemem-afm.6 (Claude marketplace/distribution install path), codemem-afm.7 (redaction hardening), codemem-afm.8 (hook no-op observability)
- Reviewed and resolved Codex comments on #128-#131; fixes were landed in the corresponding stack PRs.

I also prepared doc updates locally to align architecture/user docs with shipped multi-adapter behavior and current Claude hook scope; can ship as a focused follow-up PR after merge.
